apps/block_scout_web/assets/js/omni.js

```diff
@@ -1,199 +1,318 @@
 // local cache to re-use data
-let txMapCache = {};
+const txMapCache = {}
 
-const explorerMap = {
-  "scroll-sepolia": "https://sepolia-blockscout.scroll.io/tx/",
-  "arbitrum-goerli": "https://goerli.arbiscan.io/tx/",
-  "optimism-goerli": "https://goerli-optimism.etherscan.io/tx/",
-  "linea-goerli": "https://explorer.goerli.linea.build/tx/",
+const explorerByChain = {
+  'scroll-sepolia': 'https://sepolia-blockscout.scroll.io/tx/',
+  'arbitrum-goerli': 'https://goerli.arbiscan.io/tx/',
+  'optimism-goerli': 'https://goerli-optimism.etherscan.io/tx/',
+  'linea-goerli': 'https://explorer.goerli.linea.build/tx/',
+}
 
-  // for local test chains
-  "chain-a": "https://chain-a.etherscan.io/tx/",
-  "chain-b": "https://chain-b.etherscan.io/tx/",
-};
+const chainDisplayName = chain =>
+  chain
+    .split('-')
+    .map(word => word.charAt(0).toUpperCase() + word.slice(1))
+    .join(' ')
 
 // default api indexerUrl point to local server
-let indexerUrl = "http://localhost:8888";
+// let indexerUrl = 'http://localhost:8888'
+// in dev, use testnet explorer
+let indexerUrl = 'https://testnet-xapi.explorer.omni.network'
```

Comment:
hm how was this working without being parametrised per env?

Reply:
it's set below.  based on window hostname. which isn't ideal